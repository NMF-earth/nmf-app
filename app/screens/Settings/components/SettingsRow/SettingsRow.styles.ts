import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    paddingVertical: 14,
    paddingHorizontal: 16,
    flex: 1
  }
});
