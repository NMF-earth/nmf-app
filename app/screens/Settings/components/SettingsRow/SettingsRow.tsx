import React from "react";
import { TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import styles from "./SettingsRow.styles";
import { Text } from "../../../../components";
import { Colors } from "../../../../style";

interface Props {
  title: string;
  onPress: () => void;
}

const SettingsRow = ({ title, onPress }: Props) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <Text.Secondary bold>{title}</Text.Secondary>
      <Ionicons name={"ios-arrow-forward"} size={20} color={Colors.grey100} />
    </TouchableOpacity>
  );
};

export default SettingsRow;
