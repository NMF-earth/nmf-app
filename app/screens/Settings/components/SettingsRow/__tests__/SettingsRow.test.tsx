import React from "react";
import { Linking } from "expo";
import renderer from "react-test-renderer";
import SettingsRow from "../SettingsRow";

jest.unmock("../SettingsRow");

beforeEach(() => {
  Linking.openURL = jest.fn();
});

const props = {
  title: "title",
  onPress: () => Linking.openURL("http://nmf.earth")
};

it("SettingsRow renders correctly", () => {
  const tree = renderer.create(<SettingsRow {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("should open link", () => {
  const root = renderer.create(<SettingsRow {...props} />).root;
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  const item = root.findByType("TouchableOpacity");
  item.props.onPress();
  expect(Linking.openURL).toHaveBeenCalled();
});
