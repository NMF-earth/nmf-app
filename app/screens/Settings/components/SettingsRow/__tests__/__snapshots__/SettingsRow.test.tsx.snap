// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsRow renders correctly 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 16,
      "paddingVertical": 14,
    }
  }
>
  <Text.Secondary
    bold={true}
  >
    title
  </Text.Secondary>
  <Ionicons
    color="#111714"
    name="ios-arrow-forward"
    size={20}
  />
</TouchableOpacity>
`;
